sdcm/sct_events.py

```diff
@@ -973,4 +973,11 @@ def get_logger_event_summary():
     return output
 
 
+def stop_events_analyzer():
+    analyzer = EVENTS_PROCESSES.get('EVENTS_ANALYZER')
+    if analyzer:
+        analyzer.terminate()
+        analyzer.join(timeout=60)
+
```

Comment:
why not to add ` self.terminate()` in `sdcm.sct_events_analyzer.EventsAnalyzer.run` when handling `TestFailure` exception?
```suggestion
     class EventsAnalyzer(threading.Thread):
     ...
         def run(self):
         ...
            except TestFailure:
                self.terminate()
                self.kill_test(sys.exc_info())
```

Reply:
this doens't cover the case the kill test is being call during the `tearDown` for the the first time
we already handle this case in with `self.signal_sent`.

Reply:
why do we need `self.signal_sent` in the first place? why not to stop the event analyzer on the first critical event?

Reply:
we could, it's quite the same.
but it still won't cover a critical happening for the first time while tearDown

Reply:
I've change it to `terminate()` call on the first signal it's sending out